src/ln/peer_handler.rs

```diff
@@ -723,6 +729,24 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 						}
 						continue;
 					},
+					Event::HandleOutboundMsgGenerationError { ref node_id, ref msg } => {
+						let (mut descriptor, peer) = get_peer_for_forwarding!(node_id, {
+						});
+						peer.pending_outbound_buffer.push_back(peer.channel_encryptor.encrypt_message(&encode_msg!(msg, 17)));
+						Self::do_attempt_write_data(&mut descriptor, peer);
+					},
+					Event::DisconnectPeer { ref node_id, ref msg } => {
+						if let Some(mut descriptor) = peers.node_id_to_descriptor.remove(node_id) {
+							if let Some(mut peer) = peers.peers.remove(&descriptor) {
+								if let Some(ref msg) = *msg {
+									peer.pending_outbound_buffer.push_back(peer.channel_encryptor.encrypt_message(&encode_msg!(msg, 17)));
+									Self::do_attempt_write_data(&mut descriptor, &mut peer);
+								}
+							}
+							descriptor.disconnect_socket();
+						}
+						self.message_handler.chan_handler.peer_disconnected(&node_id, false);
```

Comment:
This should be redundant - if we dont have a descriptor for the node_id we've already disconnected it, and, thus, already called peer_disconnected.